### ISSUE TITLE
/details hardcoded / and fails on /kubecost (1.97.0)

### ISSUE THREAD

**Issue by jessegoodier** [kubecost/cost-analyzer-helm-chart#1716]
**Describe the bug**  
Customer reported and I reproduced:

kubecost 1.96.0: using ingress with a non-root (FQDN/kubecost for example) path WAI on all pages
1.97.0: /details does not respect /path, gives 404

**To Reproduce**  

1. Go to http://nginx.kubecost.xyz/197/allocations 
2. Click blue link for details of a namespace
3. See 404
4. Go to http://nginx.kubecost.xyz/196/allocations 
5. try the same
6. works

**Expected behavior**  
Worked correctly in 196

**Collect logs (please complete the following information):**
let me know if needed

sample ingress: https://github.com/kubecost/poc-common-configurations/blob/main/ingress-examples/ingress-subdir-non-root.yaml

**Comment by AjayTripathy** [comment]
@jessegoodier do you have a reproduction of this running in a cluster somewhere?

**Comment by jessegoodier** [comment]
Yes, those are my NGINX links.

**Comment by AjayTripathy** [comment]
I don't think we want a patch just yet since a workaround for now is to go directly to http://nginx.kubecost.xyz/197/details?name=kube-system&type=namespace

Bit annoying though.

**Comment by ivankube** [comment]
@nealormsbee This issue might not be fixed in the latest nightly build: http://test-kubecost.kbrwn.xyz/kubecost/allocations clicking on any namespace results in [404](http://test-kubecost.kbrwn.xyz/details?name=agones-system&type=namespace)

❯ kubectl get deploy kubecost-cost-analyzer -o yaml -n kubecost | grep frontend
        image: gcr.io/kubecost1/frontend-nightly:188b33b

**Comment by aschi1** [comment]
We are having the same problem with kubecost exposed on FQDN/kubecost. Additionally, after OIDC login via Keycloak I am redirected to FQDN/login and not FQDN/kubecost/login (we are providing redirect uri in the keycloak request).

It would help us immensely f this could be fixed. Thanks!

**Comment by AjayTripathy** [comment]
@jjarrett21 was what @ivankube said missed in https://github.com/kubecost/cost-analyzer-frontend/pull/1123 or did that build just not make it to that cluster/nightly? Can you take a quick look since neal is out?

**Comment by AjayTripathy** [comment]
Hey @ivankube can you try to deploy the latest nightly and confirm what you're seeing?

**Comment by jessegoodier** [comment]
will leave this up permanently:
https://nightly-subpath.kubecost.xyz/kubecost/
seems broken still?

**Comment by ivankube** [comment]
@AjayTripathy I tested with the latest nightly frontend[ image]( gcr.io/kubecost1/frontend-nightly:e220762) and got the same error as Jesse - 404:

**Comment by ddl-wasanthag** [comment]
+1 @ivankube I see still see the same broken behavior.

**Comment by AjayTripathy** [comment]
@jjarrett21 can you take another look?

**Comment by ivankube** [comment]
@jjarrett21 please use non-root ingress for repro.

**Comment by ivankube** [comment]
It seems like `biancaburtoiu/jj-new-details-path` fixed the issue. I don't receive 404 anymore. I'd wait for another confirmation though.

**Comment by Adam-Stack-PM** [comment]
Jesse tested and confirmed the fix.

**Comment by jessegoodier** [comment]
Nice work team!

**Comment by AjayTripathy** [comment]
(FYI this will be fixed in v1.98-rc.2+)
